cmd/juju/application/bundle_test.go

```diff
@@ -244,11 +244,40 @@ func (s *BundleDeploySuite) assertDeployBundleLocalPathInvalidSeriesWithForce(c
 		args = append(args, "--force")
 	}
 	err = s.runDeploy(c, args...)
-	if force {
-		c.Assert(err, gc.ErrorMatches, "cannot deploy bundle: base: ubuntu@12.10/stable not supported")
-	} else {
-		c.Assert(err, gc.ErrorMatches, `cannot deploy bundle:.*base "ubuntu@12.10" not supported by charm.*`)
-	}
+	c.Assert(err, gc.ErrorMatches, "cannot deploy bundle: base: ubuntu@12.10/stable not supported")
+}
+
+func (s *BundleDeploySuite) TestDeployBundleLocalPathInvalidJujuSeries(c *gc.C) {
+	dir := c.MkDir()
+	charmDir := testcharms.RepoWithSeries("bionic").ClonedDir(dir, "dummy")
+
+	dummyURL := charm.MustParseURL("local:dummy-1")
+	withLocalCharmDeployable(s.fakeAPI, dummyURL, charmDir, false)
+	withLocalBundleCharmDeployable(
+		s.fakeAPI, dummyURL, base.MustParseBaseFromString("ubuntu@12.10"),
+		charmDir.Meta(), charmDir.Manifest(), false,
+	)
+	s.fakeAPI.Call("CharmInfo", "local:dummy-1").Returns(
+		&apicommoncharms.CharmInfo{
+			URL:  "local:dummy",
+			Meta: &charm.Meta{Name: "dummy", Series: []string{"quantal", "jammy"}},
+		},
+		error(nil),
+	)
+
+	path := filepath.Join(dir, "mybundle")
+	data := `
+        series: focal
+        applications:
+            dummy:
+                charm: ./dummy
+                num_units: 1
+    `
+	err := os.WriteFile(path, []byte(data), 0644)
+	c.Assert(err, jc.ErrorIsNil)
+
+	err = s.runDeploy(c, path)
+	c.Assert(err, gc.ErrorMatches, `cannot deploy bundle: base "ubuntu@20.04/stable" is not supported, supported bases are: .*`)
```

Comment:
The above changes are a result of `NewCharmAtPath` being decoupled from selecting a base. This has lead to a slight re-ordering of our checks whilst deploying